### PR TITLE
E2e improvements

### PR DESCRIPTION
**Pull request by flacatus**
### What does this PR do?
This script improve old one by adding new commands used in PROW CI. The script and e2e tests was tested successfully in OCP 4.4 and OCP 4.5 in PROW.

Test results: 

- OC 4.4 : https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/9605/rehearse-9605-pull-ci-devfile-devworkspace-operator-e2e_improvements-v4-openshift-web-terminal-e2e/5

- OC 4.5 : https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/9605/rehearse-9605-pull-ci-devfile-devworkspace-operator-e2e_improvements-v5-openshift-web-terminal-e2e/5

### CODE REVIEW

File: .cico/cico_openshift_e2e.sh [hunk 1]
```diff
@@ -9,41 +9,39 @@
...
+export GIT_COMMITTER_NAME="CI BOT"
+export GIT_COMMITTER_EMAIL="ci_bot@notused.com"
 
-# Exported to current env
-export KUBECONFIG=${KUBECONFIG:-"${DEFAULT_INSTALLER_ASSETS_DIR}/auth/kubeconfig"}
+# Check if operator-sdk is installed and if not install operator sdk in $GOPATH/bin dir
+if ! hash operator-sdk 2>/dev/null; then
+    mkdir -p $GOPATH/bin
+    export PATH="$PATH:$(pwd):$GOPATH/bin"
+    RELEASE_VERSION=v0.17.0
```
Comment (sleshchenko) [inline .cico/cico_openshift_e2e.sh:24]: Let's rename it to OPERATOR_SDK_VERSION?

File: .cico/cico_openshift_e2e.sh [hunk 2]
```diff
@@ -9,41 +9,39 @@
...
+# Add kubernetes-api CRDS
+make update_devworkspace_crds
 
-set +x
-# Login as admin user
-oc login -u $KUBEADMIN_USER -p $KUBEADMIN_PASSWORD --insecure-skip-tls-verify
-set -x
+# Install go modules
+go mod tidy
+go mod vendor
```
Comment (sleshchenko) [inline .cico/cico_openshift_e2e.sh:38]: To make vendor folder used, you probably need to add `-mod=vendor` to go test command. But It's OK just download dependencies to be inlined in build in Dockerfile
```suggestion
go mod download
```

File: .cico/cico_openshift_e2e.sh [hunk 3]
```diff
@@ -9,41 +9,39 @@
...
-export KUBEADMIN_PASSWORD=$(cat $KUBEADMIN_PASSWORD_FILE)
+# Add kubernetes-api CRDS
+make update_devworkspace_crds
 
-set +x
-# Login as admin user
-oc login -u $KUBEADMIN_USER -p $KUBEADMIN_PASSWORD --insecure-skip-tls-verify
-set -x
+# Install go modules
+go mod tidy
```
Comment (sleshchenko) [inline .cico/cico_openshift_e2e.sh:37]: Could you elaborate why it's needed?

File: .cico/cico_openshift_e2e.sh (hunk 1 above)
Comment (flacatus) [reply .cico/cico_openshift_e2e.sh:24]: Sure.

File: .cico/cico_openshift_e2e.sh (hunk 3 above)
Comment (flacatus) [reply .cico/cico_openshift_e2e.sh:37]: For some reason after doing `go mod download` in prow still not found the necessary modules. If I used `go mod tidy` and `go mod vendor` works...

**Pull request merged**